File: services/agent-ts/src/routes/whoami.ts
/**
 * `/v1/whoami` — auth smoke test. Echoes the verified JWT claims
 * (minus secrets) so an operator can confirm the JWT secret + the
 * Authorization-header plumbing work end-to-end before any real
 * endpoints exist.
 *
 * Mirrors `services/agent-py`'s `/v1/whoami` exactly.
 */

import { Hono } from "hono"

import type { AuthVars } from "../middleware/auth"
import { requireAuth } from "../middleware/auth"

export const whoamiRoutes = new Hono<{ Variables: AuthVars }>()

whoamiRoutes.get("/v1/whoami", requireAuth, (c) => {
  const claims = c.get("claims")
  return c.json({
    user_id: typeof claims.sub === "string" ? claims.sub : null,
    role: typeof claims.role === "string" ? claims.role : null,
  })
})

File: services/agent-py/src/agent_py/routers/whoami.py
"""Auth smoke test endpoint. Verifies the JWT secret and
Authorization-header plumbing work end-to-end before any real
endpoints exist.
"""

from __future__ import annotations

from typing import Annotated

from fastapi import APIRouter, Depends
from pydantic import BaseModel

from ..auth import get_current_user

router = APIRouter(tags=["auth"])


class WhoAmIResponse(BaseModel):
    user_id: str | None
    role: str | None


@router.get("/v1/whoami", response_model=WhoAmIResponse)
def whoami(
    claims: Annotated[dict[str, object], Depends(get_current_user)],
) -> WhoAmIResponse:
    """Auth smoke test — echoes the verified claims (minus secrets).

    Useful during Phase 0 deployment to confirm the JWT secret and
    the Authorization-header plumbing work end-to-end before any
    real endpoints exist. Deliberately under `/v1` so the prefix
    convention exists from day one.
    """
    sub = claims.get("sub")
    role = claims.get("role")
    return WhoAmIResponse(
        user_id=str(sub) if sub is not None else None,
        role=str(role) if role is not None else None,
    )
